### PR TITLE
Introduce Spotbugs plugin

### PR DESCRIPTION
**Pull request by brfrn169**
This PR introduces the `Spotbugs` plugin that uses static analysis to look for bugs in Java code. Also, I fixed some Spotbugs errors in this PR.

Honestly, I'm not sure it's a good idea for us to use this plugin because it shows some false positives, and we need to use `SuppressFBWarnings` annotation to remove them, which might be annoying. But of course, I think it brings many benefits to us. So I would like to discuss whether or not we use this plugin here.

Besides this plugin, I'm thinking that we can use the `errorprone` plugin, as well:
https://errorprone.info/
https://github.com/tbroyer/gradle-errorprone-plugin

Please take a look!

### CODE REVIEW

**Review by Torch3333 — COMMENTED** [review]
I think it's a good idea. I agree that it may become troublesome for some of the rules that cause a lot of false positive but I wonder if it will often trigger, personally I think we  could try this tool for some time and then see if it's an hindrance or not. Also, I will need to try the Intellij plugin as well.

Besides, thanks for all the fixes 🙂

**Review by feeblefakie — COMMENTED** [review]
Thank you! The code looks good but I left one question.
It's definitely nice to have so let's see how it goes.
I'll merge after we discuss it tomorrow.

File: core/src/main/java/com/scalar/db/storage/cassandra/ClusterManager.java [hunk 1]
```diff
@@ -1,7 +1,5 @@
 package com.scalar.db.storage.cassandra;
```
[the remark this answers is not in the document]
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/storage/cassandra/ClusterManager.java:1]: It was the one to fix a `synchronized` related warning. Let's discuss it later.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you 🙂

**Pull request merged**